ROX-24433: (B)ANP integration

shireenf-ibm opened the pull request.

shireenf-ibm committed:
bump netpol-analyzer to v1.3.0+ changes to be compatible with new version:
1. updating expected output file with new line at eof
2. comment out in bats tests an unwrap error that returns by metav1.LabelSelectorAsSelector(ruleSelector)  (can be resolve in v1.3.1)

shireenf-ibm committed:
Merge branch 'master' into rox_24433_b_anp_integration

shireenf-ibm commented:
hi @adisos,
this branch is for integrating (B)ANP into roxctl

[first commit :](https://github.com/shireenf-ibm/stackrox/pull/9/commits/e607a5b2ab27ecf8dcf5f0946748437bd8d8687b) bump netpol-analyzer to v1.3.0; I found that this bump will cause tests failure of the `roxctl` :
so made changes to be compatible with new version:

1. on our latest netpol-analyzer version we have updated textual output with a new line at the EOF;  (so updated expected output `roxctl/netpol/connectivity/map/testdata/acs-security-demos/output.txt` with new line at EOF)
2. two bats tests failed since in the new version v1.3.0 an error title was removed, in this commit commented out them (I opened [PR](https://github.com/np-guard/netpol-analyzer/pull/497) to return it- so this can be resolved with (v1.3.1) if we want to keep the error's title


next commit will be with new bats tests containing (b)anps

vladbologa committed:
fix(sensor): use correct annotations for Sensor TLS secrets (#14023)

pedrottimark committed:
ROX-27742: Add search filter for EPSS Probability (#13949)

bradr5 committed:
ROX-27531: Add external ips table (#13998)

adisos commented:
> hi @adisos, this branch is for integrating (B)ANP into roxctl
> 
> [first commit :](https://github.com/shireenf-ibm/stackrox/pull/9/commits/e607a5b2ab27ecf8dcf5f0946748437bd8d8687b) bump netpol-analyzer to v1.3.0; I found that this bump will cause tests failure of the `roxctl` : so made changes to be compatible with new version:
> 
> 1. on our latest netpol-analyzer version we have updated textual output with a new line at the EOF;  (so updated expected output `roxctl/netpol/connectivity/map/testdata/acs-security-demos/output.txt` with new line at EOF)
> 2. two bats tests failed since in the new version v1.3.0 an error title was removed, in this commit commented out them (I opened [PR](https://github.com/np-guard/netpol-analyzer/pull/497) to return it- so this can be resolved with (v1.3.1) if we want to keep the error's title
> 
> next commit will be with new bats tests containing (b)anps

sounds good.
consider also specify in documentation (within roxctl/netpol/connectivity/map) the supported policy resources.

shireenf-ibm committed:
bump netpol-analyzer to v1.3.1 (and bats tests changes)

dependabot[bot] committed:
chore(deps): bump com.slack.api:slack-api-client from 1.45.1 to 1.45.2 in /qa-tests-backend (#14037)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/service/s3 from 1.74.1 to 1.75.0 (#14039)

msugakov committed:
ROX-22250: Switch to sanitizing version of determine-image-tag (#13995)

shireenf-ibm committed:
tiny doc update

msugakov committed:
ROX-27764: Stop providing PipelineRun in AppStudio's label (#14011)

dependabot[bot] committed:
chore(deps): bump github.com/spf13/pflag from 1.0.6-0.20210604193023-d5e0c0615ace to 1.0.6 (#14040)

porridge committed:
feat(helm): only specify `imagePullSecrets:` when not empty (#14025)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/service/ecr from 1.38.7 to 1.39.0 (#14041)

dashrews78 committed:
chore(owners): add search to CWF (#14044)

davdhacs committed:
ROX-25270: on osd-aws re-disable k8srbac test (#14033)

shireenf-ibm committed:
adding bats tests with (b)anp

lvalerom committed:
fix(log): Rate limit the log for drop item (#14046)

ajheflin committed:
ROX-26586: Remove search tags from ListAlert (#13891)

ajheflin committed:
ROX-27782: Make pruning stop after context errors (#13965)

shireenf-ibm committed:
fixing eof empty line

shireenf-ibm committed:
fixing eof empty line

shireenf-ibm commented:
added bats tests for both connectivity map and diff (with (B)ANPs);
for connectivity map : added two tests (one with exposure) - each tested with multiple output formats
and on test (two dirs) for connectivity diff - with multiple output formats

lvalerom committed:
feat(cancel-scan): Stop scan request if the context is canceled (#14024)

dcaravel committed:
ROX-27146: SBOM Scanner V4 Matcher API (#13756)

dcaravel committed:
fix(ci): due to diff CDN behavior (#14052)

dvail committed:
ROX-27822: Add horizontal subnav support and impl for VM (#13997)

pedrottimark committed:
ROX-27894: Add SortAggregate options for EPSS probability (#14048)

bradr5 committed:
ROX-27862: Update policy activation state (#14031)

RTann committed:
chore(CODEOWNERS): update Scanner team ownership (#14051)

shireenf-ibm committed:
format

shireenf-ibm committed:
resolve misc-checks (no trailing whitespaces)

shireenf-ibm committed:
avoiding large files error

shireenf-ibm commented:
the bats-tests files with the new tests are considered large files 
![Screenshot 2025-02-02 145113](https://github.com/user-attachments/assets/31a3f7cf-8006-4051-8124-e5f1ce05c87e)

in order to avoid this error - used partial expected output match check (instead of full) [last commit](https://github.com/shireenf-ibm/stackrox/pull/9/commits/8d32828ea09905d53d68ba87bdbcb5f910aa82a1)

shireenf-ibm committed:
avoid trailing whitespaces

stehessel committed:
ROX-25373: default values if no config is found (#14019)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2 from 1.34.0 to 1.36.0 (#14078)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/service/ecr from 1.39.0 to 1.40.1 (#14079)

lvalerom committed:
ROX-27904: Use fake ticker in test (#14068)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/credentials from 1.17.55 to 1.17.57 (#14075)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/feature/s3/manager from 1.17.54 to 1.17.57 (#14077)

dvail committed:
ROX-27823: Update routing for VM result views (#14013)

rhacs-bot committed:
chore(collector): Update COLLECTOR_VERSION (#13898)

rhacs-bot committed:
chore(scanner): Update SCANNER_VERSION (#14074)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/service/securityhub from 1.55.6 to 1.55.8 (#14076)

dependabot[bot] committed:
chore(deps): bump com.google.api.grpc:proto-google-common-protos from 2.50.1 to 2.51.0 in /qa-tests-backend (#14055)

dvail committed:
ROX-27825: Replace images w/o vulns dropdown with view (#14049)

ksurabhi91 committed:
ROX-27529: Add epss column to vuln report based on filter (#13950)

daynewlee committed:
ROX-27637: enrich EPSS in vulnerability report (#13779)

Co-authored-by: Ross Tannenbaum <rtannenb@redhat.com>

daynewlee committed:
fix: matcher vuln URL (#14090)

dependabot[bot] committed:
chore(deps): bump golang.stackrox.io/grpc-http1 from 0.3.13 to 0.4.0 (#14095)

dependabot[bot] committed:
chore(deps): bump github.com/openshift-online/ocm-sdk-go from 0.1.456 to 0.1.458 (#14093)

dependabot[bot] committed:
chore(deps): bump com.google.code.gson:gson from 2.11.0 to 2.12.1 in /qa-tests-backend (#14092)

msugakov committed:
build: Switch Konflux UI builds from NodeJS 18 to 20 (#14086)

red-hat-konflux[bot] committed:
chore(deps): update konflux references (#14036)

mtodor committed:
ROX-27818: Ensure api Stop is properly called and finished (#14009)

dvail committed:
ROX-27828: Replace Violations dropdown with nav (#14067)

jschnath committed:
ROX-26595: Test watcher suite timeout (#13991)

parametalol committed:
ROX-27902: telemetry campaign headers problem (#14063)

RTann committed:
chore(CODEOWNERS): add more Scanner team ownership (#14088)

bradr5 committed:
ROX-27639: Add side panel for single external entity (#14034)

pedrottimark committed:
ROX-27423: Enable ROX_EPSS_SCORE feature flag (#14032)

dependabot[bot] committed:
chore(deps): bump the operator-framework group across 1 directory with 2 updates (#13387)

dependabot[bot] committed:
chore(deps): bump golang.org/x/sys from 0.29.0 to 0.30.0 (#14119)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/credentials from 1.17.57 to 1.17.58 (#14121)

dependabot[bot] committed:
chore(deps): bump github.com/kudobuilder/kuttl from 0.20.0 to 0.21.0 in /operator/tools/kuttl (#14112)

dependabot[bot] committed:
chore(deps): bump golang.org/x/net from 0.30.0 to 0.33.0 in /operator/tools/operator-sdk (#14122)

dependabot[bot] committed:
chore(deps): bump golang.org/x/text from 0.21.0 to 0.22.0 (#14118)

mtodor committed:
ROX-27931: Revert require on api Stop call (#14123)

dvail committed:
ROX-27826: Update description and tooltips for VM results page (#14085)

SimonBaeumer committed:
docs: Add service principal command to Sentinel notifier (#14102)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/feature/s3/manager from 1.17.57 to 1.17.58 (#14117)

janisz committed:
chore(pg): add comment about IN operator (#14127)

Signed-off-by: Tomasz Janiszewski <tomek@redhat.com>

janisz committed:
ROX-27937: retry on error (#14126)

Signed-off-by: Tomasz Janiszewski <tomek@redhat.com>

JoukoVirtanen committed:
feat(performance-tests): Add k6 integration to CI pipeline jv (#14069)

Co-authored-by: Mladen Todorovic <mtodor@redhat.com>
Co-authored-by: Mladen Todorovic <mtodor@gmail.com>
Co-authored-by: Tom Martensen <tmartens@redhat.com>

dependabot[bot] committed:
chore(deps): bump golang.org/x/mod from 0.22.0 to 0.23.0 (#14115)

dependabot[bot] committed:
chore(deps): bump ch.qos.logback:logback-classic from 1.5.6 to 1.5.16 in /qa-tests-backend (#14113)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump golang.org/x/time from 0.9.0 to 0.10.0 (#14120)

dvail committed:
fix(UI): Update links to support split of Platform CVEs (#14110)

dvail committed:
ROX-27936: Fixes lost handling of quotes in filename payload (#14128)

dcaravel committed:
ROX-27772: SBOM fix scan time update edge case (#14089)

dependabot[bot] committed:
chore(deps): bump github.com/go-git/go-git/v5 from 5.11.0 to 5.13.0 in /operator/tools/operator-sdk (#14131)

pedrottimark committed:
ROX-27743: Add info popover for EPSS probability column (#14070)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/service/sts from 1.33.13 to 1.33.14 (#14143)

porridge committed:
tests: improve ticket titles for failed sensor and collector check (#14124)

tommartensen committed:
chore(ci): Bump upgrade test target version to 4.6.2 (#14101)

Stringy committed:
ROX-18384: removes the slim collector distinction (#13350)

All installation methods are updated to either ignore or translate the 
slim mode, so that regular images are used throughout. supporting
code has been updated to reflect this, and duplicate variables for 
slim/full collector have been consolidated into single variable alternatives.

Co-authored-by: Dmitrii Dolgov <9erthalion6@gmail.com>
Co-authored-by: Piotr Rygielski <114479+vikin91@users.noreply.github.com>
Co-authored-by: Moritz Clasmeier <111092021+mclasmeier@users.noreply.github.com>

janisz committed:
perf(sensor): no alloc trace writer (#14064)

Signed-off-by: Tomasz Janiszewski <tomek@redhat.com>

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/feature/ec2/imds from 1.16.27 to 1.16.28 (#14140)

dependabot[bot] committed:
chore(deps): bump org.apache.groovy:groovy-bom from 4.0.24 to 4.0.25 in /qa-tests-backend (#14139)

mtodor committed:
ROX-27931: Revert stopper state propagation (#14151)

dependabot[bot] committed:
chore(deps): bump golang.org/x/term from 0.28.0 to 0.29.0 (#14142)

ajheflin committed:
ROX-27700: Restore cluster datastore tests (#14066)

dvail committed:
chore(ui): Enable the platform CVE split feature flag (#14133)

ksurabhi91 committed:
ROX-27950: Fix rep header (#14138)

dcaravel committed:
ROX-27606: add SBOM gen to changelog (#14134)

sachaudh committed:
ROX-27745: Adding external ips table to deployment side panel (#14109)

dependabot[bot] committed:
chore(deps): bump fabric8Version from 7.0.1 to 7.1.0 in /qa-tests-backend (#14161)

mclasmeier committed:
ROX-27911: Lower CPU request for init-tls-certs container (#14099)

Co-authored-by: Moritz Clasmeier <mclasmeier@redhat.com>

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/service/s3 from 1.75.3 to 1.76.0 (#14163)

parametalol committed:
ROX-27781: periodic reload telemetry config (#13963)

Stringy committed:
feat(networkgraph): metadata endpoint and external flows adjustments (#13861)

Contains two enhancements to the network graph service, for external entities and their associated flows. These have been added to directly support new external IP UI features.

GetExternalNetworkFlows (v1/networkgraph/cluster/{cluster_id}/externalentities/{entity_id}/flows) has been changed to return a list of flows for a given external network entity.

GetExternalNetworkFlowsMetadata (v1/networkgraph/cluster/{cluster_id}/externalentities/metadata) has been added and returns a list of entities along with a count of the number of flows that communicate with them.

Co-authored-by: Mauro Ezequiel Moltrasio <mmoltras@redhat.com>

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/config from 1.29.5 to 1.29.6 (#14146)

dvail committed:
chore(ui): Add descriptions for Violation page states (#14087)

Stringy committed:
ROX-27971: removes ROX_NETWORK_GRAPH_EXTERNAL_IPS feature flag (#14168)

Signed-off-by: Giles Hutton <ghutton@redhat.com>

stehessel committed:
chore(image-signatures): improve UX of the integration form (#14150)

dependabot[bot] committed:
chore(deps): bump github.com/openshift-online/ocm-sdk-go from 0.1.458 to 0.1.459 (#14166)

rhybrillou committed:
test: trace the workflow of DeclarativeConfigTest tests (#14010)

erthalion committed:
ROX-27438: Add db metrics to telemetry (#14104)

There are few database related metrics, that are not exposed outside of ACSCS:

* Database size, which is an important factor for making decisions about
  strategies for upgrade, HA, etc.

* PostgreSQL version, which is necessary to monitor the major version upgrade
  status.

* Whether the database is external, which gives information about how many
  users prefer to go with out of the box solution.

Make this data availabe via telemetry as well.

Co-authored-by: Michaël <smartptr@gmail.com>

dependabot[bot] committed:
chore(deps): bump google.golang.org/protobuf from 1.36.4 to 1.36.5 (#14165)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/feature/s3/manager from 1.17.58 to 1.17.60 (#14164)

janisz committed:
ROX-27973: use pgx.CollectRows (#14152)

Signed-off-by: Tomasz Janiszewski <tomek@redhat.com>

stehessel committed:
ROX-27649: Azure workload identity support for Microsoft Sentinel (#14027)

parametalol committed:
fix(telemetery): mutex refactoring (#14153)

pedrottimark committed:
ROX-27967: Select max epssProbability from distroTuples (#14159)

stehessel committed:
ROX-27978: image signature e2e test (#14170)

vikin91 committed:
ROX-27719: RHCOS Node scanning V4 final touches (#14100)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/service/securityhub from 1.55.8 to 1.55.9 (#14141)

ksurabhi91 committed:
ROX-27784: remove cluster flag from http handler and roxctl api (#13951)

RTann committed:
ROX-27253: inconsistent RHSA data (#13523)

stehessel committed:
ROX-21529: Azure workload identity for ACR (#13635)

dcaravel committed:
ROX-23735: Create integration for OCP Global Pull Secret (#13582)

dependabot[bot] committed:
chore(deps): bump github.com/aws/aws-sdk-go-v2/service/ecr from 1.40.1 to 1.41.0 (#14184)

dependabot[bot] committed:
chore(deps): bump io.kubernetes:client-java from 22.0.0 to 22.0.0-legacy in /qa-tests-backend (#14186)

dependabot[bot] committed:
chore(deps): bump github.com/grpc-ecosystem/grpc-gateway/v2 from 2.26.0 to 2.26.1 (#14183)

rhybrillou committed:
tests: refresh queries for main and vulnerability dashboards (#14167)

aaa5kameric committed:
ROX-25844: Add Labels to all TLS Secrets (#13162)

Stringy committed:
ROX-27612: Removes Collector slim references from tekton pipeline (#14169)

lvalerom committed:
feat(sensor): Make the responses channel buffered (#13925)

dependabot[bot] committed:
chore(deps): bump golang.org/x/crypto from 0.32.0 to 0.33.0 (#14185)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

vikin91 committed:
ROX-27719: Enable FF for RHCOS node scanning with scanner V4  (#14154)

Includes: Promoting an boolean setting variable to a feature-flag

rhybrillou committed:
chore(dependabot): group dependabot PRs for aws-sdk-go-v2 (#14177)

vikin91 committed:
fix(sensor): Avoid unnecessary image scan attempts (#13935)

lvalerom committed:
feat(metrics): Add metrics to the scan logic in the detector (#14190)

rhacs-bot committed:
chore(scanner): Update SCANNER_VERSION (#14182)

bradr5 committed:
chore(ui): Fix network graph routing on mount (#14132)

vladbologa committed:
ROX-27597: Unit tests for cluster sensor capabilities (#14171)

shireenf-ibm committed:
Merge branch 'master' into rox_24433_b_anp_integration

shireenf-ibm committed:
Merge branch 'master' into rox_24433_b_anp_integration

adisos commented on roxctl/netpol/connectivity/diff/README.md:
```diff
@@ -2,7 +2,7 @@
 ## About
 
-The static network policy connectivity diff is a tool that analyzes two sets of Kubernetes manifests, including network policies.
+The static network policy connectivity diff is a tool that analyzes two sets of Kubernetes manifests, including network policies (i.e. NetworkPolicy, AdminNetworkPolicy and BaselineAdminNetworkPolicy objects).
```
```suggestion
The static network policy connectivity diff is a tool that analyzes two sets of Kubernetes manifests, including network policies (i.e. NetworkPolicy, AdminNetworkPolicy and BaselineAdminNetworkPolicy resources).
```

adisos commented on roxctl/netpol/connectivity/map/README.md:
```diff
@@ -1,7 +1,7 @@
 
 ## About
-The static network policy analyzer is a tool that analyzes Kubernetes network policies.
+The static network policy analyzer is a tool that analyzes Kubernetes network policies (i.e. NetworkPolicy, AdminNetworkPolicy and BaselineAdminNetworkPolicy objects).
```
```suggestion
The static network policy analyzer is a tool that analyzes Kubernetes network policies (i.e. NetworkPolicy, AdminNetworkPolicy and BaselineAdminNetworkPolicy resources).
```

adisos commented on roxctl/netpol/connectivity/map/testdata/acs-security-demos/output.txt:
```diff
@@ -11,4 +11,4 @@ frontend/webapp[Deployment] => backend/shipping[Deployment] : TCP 8080
 {ingress-controller} => frontend/asset-cache[Deployment] : TCP 8080
-{ingress-controller} => frontend/webapp[Deployment] : TCP 8080
\ No newline at end of file
+{ingress-controller} => frontend/webapp[Deployment] : TCP 8080
```
what has changed here?

shireenf-ibm replied:
`\n` at the end of the output (the cursor will be on head of next line instead of after `8080`
(added because of the `Issue 424 : textual output small fix: generating output with empty line at EOF` in netpol ananlyzer in last version)

shireenf-ibm replied:
mentioned it also [here](https://github.com/shireenf-ibm/stackrox/pull/9#issuecomment-2622564774)

adisos commented on tests/roxctl/bats-tests/local/roxctl-netpol-connectivity-map-development.bats:
```diff
@@ -437,19 +437,7 @@ payments/gateway[Deployment] => payments/visa-processor[Deployment] : TCP 8080'
-payments/gateway[Deployment] => payments/mastercard-processor[Deployment] : TCP 8080
-payments/gateway[Deployment] => payments/visa-processor[Deployment] : TCP 8080
-{ingress-controller} => frontend/asset-cache[Deployment] : TCP 8080
+  partial_expected_output='{ingress-controller} => frontend/asset-cache[Deployment] : TCP 8080
```
existing tests from main had to change?

shireenf-ibm replied:
no, just the `\n` at EOF (the green line existed also in the original output);
the change here was with the name of the variable `partial_expected_output`  + shortened the expected output data to partial instead of full to avoid getting large files - see [comment](https://github.com/shireenf-ibm/stackrox/pull/9#issuecomment-2629415274)

shireenf-ibm replied:
here made it patial output by removing some of the connlist output and focusing on the exposure output

shireenf-ibm replied:
since the files are not large now (after removing new tests with formats other than txt) i can revert changes to old tests; to do that?

adisos replied:
yes, please

shireenf-ibm replied:
done

adisos commented on tests/roxctl/bats-tests/local/roxctl-netpol-connectivity-map-development.bats:
```diff
@@ -605,6 +565,253 @@ hello-world/workload-a[Deployment] is not protected on Egress'
   assert_output "$normalized_expected_output"
 }
 
+@test "roxctl-development netpol connectivity map generates connlist for input resources with admin network policies" {
```
I think it should be sufficient to just add the `txt` format for the added ANP/BANP tests  (both here and in diff).
There are other tests focusing on these other formats.

shireenf-ibm replied:
done

shireenf-ibm committed:
Update roxctl/netpol/connectivity/diff/README.md

Co-authored-by: Adi Sosnovich <82078442+adisos@users.noreply.github.com>

shireenf-ibm committed:
Update roxctl/netpol/connectivity/map/README.md

Co-authored-by: Adi Sosnovich <82078442+adisos@users.noreply.github.com>

shireenf-ibm committed:
removing new tests with other that txt formats

shireenf-ibm committed:
revert changes to old tests

adisos commented on tests/roxctl/bats-tests/local/roxctl-netpol-connectivity-map-development.bats:
```diff
@@ -605,6 +605,69 @@ hello-world/workload-a[Deployment] is not protected on Egress'
+  assert_output "$normalized_expected_output"
+}
+
+@test "roxctl-development netpol connectivity map generates exposure report for input resources with admin network policies" {
```
the test with exposure analysis can also be removed

shireenf-ibm replied:
done

shireenf-ibm committed:
removing exposure with anps test
